packages/fiber/src/native/index.tsx

```diff
@@ -0,0 +1,217 @@
+import { View, PixelRatio } from 'react-native'
+import { ExpoWebGLRenderingContext } from 'expo-gl'
+
+const roots = new Map<ExpoWebGLRenderingContext & WebGLRenderingContext, Root>()
+const { invalidate, advance } = createLoop(roots)
+const { reconciler, applyProps } = createRenderer(roots, getEventPriority)
+
+export type RenderProps<TView extends View> = Omit<StoreProps, 'gl' | 'events' | 'size'> & {
+  gl?: THREE.WebGLRenderer | Partial<THREE.WebGLRendererParameters>
+  events?: (store: UseStore<RootState>) => EventManager<TView>
+  size?: Size
+  onCreated?: (state: RootState) => void
+}
+
+const createRendererInstance = (
+  gl: THREE.WebGLRenderer | Partial<THREE.WebGLRendererParameters> | undefined,
+  context: ExpoWebGLRenderingContext & WebGLRenderingContext,
+): THREE.WebGLRenderer => {
+  // If a renderer is specified, return it
+  if (isRenderer(gl as THREE.WebGLRenderer)) return gl as THREE.WebGLRenderer
+
+  // Create canvas shim
+  const canvas = {
+    width: context.drawingBufferWidth,
+    height: context.drawingBufferHeight,
+    style: {},
+    addEventListener: (() => {}) as any,
+    removeEventListener: (() => {}) as any,
+    clientHeight: context.drawingBufferHeight,
+  } as HTMLCanvasElement
+
+  // Create renderer and pass our canvas and its context
+  const renderer = new THREE.WebGLRenderer({
+    powerPreference: 'high-performance',
+    antialias: true,
+    alpha: true,
+    ...(gl as any),
+    canvas,
+    context,
+  })
+
+  // Bind render to RN bridge
+  const renderFrame = renderer.render.bind(renderer)
+  renderer.render = (scene: THREE.Scene, camera: THREE.Camera) => {
+    renderFrame(scene, camera)
+    context.endFrameEXP()
+  }
+
+  return renderer
+}
+
+function createRoot(context: ExpoWebGLRenderingContext & WebGLRenderingContext) {
+  return {
+    render: (element: React.ReactNode) => render(element, context),
+    unmount: () => unmountComponentAtNode(context),
+  }
+}
+
+function render<TView extends View>(
+  element: React.ReactNode,
+  context: ExpoWebGLRenderingContext & WebGLRenderingContext,
+  { dpr = PixelRatio.get(), gl, size = { width: 0, height: 0 }, events, onCreated, ...props }: RenderProps<TView> = {},
+): UseStore<RootState> {
+  let root = roots.get(context)
+  let fiber = root?.fiber
+  let store = root?.store
+  let state = store?.getState()
+
+  if (fiber && state) {
+    const lastProps = state.internal.lastProps
+
+    // When a root was found, see if any fundamental props must be changed or exchanged
+
+    // Check pixelratio
+    if (dpr !== undefined && !is.equ(lastProps.dpr, dpr)) state.setDpr(dpr)
+    // Check size
+    if (!is.equ(lastProps.size, size)) state.setSize(size.width, size.height)
+
+    // For some props we want to reset the entire root
+
+    // Changes to the color-space
+    const linearChanged = props.linear !== lastProps.linear
+    if (linearChanged) {
+      unmountComponentAtNode(context)
+      fiber = undefined
+    }
+  }
```

Comment:
This will have to be updated according to https://github.com/pmndrs/react-three-fiber/pull/1707 once that's merged.